### DISCUSSION TITLE
FUSE: Restrictions/Limitations for UNC-Path option?

### DISCUSSION THREAD

**Discussion by infeo** [winfsp/winfsp#500] [Q&A]
Hey,

the FUSE api of WinFSP offers the`VolumePrefix` option, to mount the filesystem as a network drive, see
https://github.com/winfsp/winfsp/blob/c61679a35d041d843173fa3b2eba106b5ab7b01f/src/dll/fuse/fuse.c#L103-L106
and
https://github.com/winfsp/winfsp/blob/6fb72555d33d228736b8156cb919edeaae4da150/src/dll/fuse/fuse.c#L640-L641

Mounting the filesystem, it is then accessible over the assigned drive letter _and_ the UNC path. According to the help output, the path should be of form `/server/share`. If for the "share" part a path with more than one section is used, the filesystem is still mounted and accessible via its assigned drive letter, but you cannot access it anymore via its UNC path.

For example, if I mount the filesystem with option `-oVolumePrefix=/localhost/myShare/foobar`, i cannot access the filesystem with `\\localhost\myShare\foobar`.

**Comment by billziss-gh** [comment]
When choosing UNC paths you should avoid using prefixes that are already meaningful. The name `localhost` already has a well defined meaning, so it is likely that one of the built-in network redirectors is attempting to handle it.

 Use a name like `/cryptomator/myShare/foobar` and it should work.

**Reply by infeo** [reply]
I tried this with the memfs-fuse3 filesystem: `.\memfs-fuse3-x64.exe -ouid=-1 -ogid=-1 -oUNC=/some/test/path`

Changing director to the network share failed:
```
PS C:\Users\Arbeit> cd \\some\test\path
Set-Location: The network path was not found. : '\\some\test'
```

**Reply by infeo** [reply]
Also ensured, that WinFsp is the first in the network provider order

**Comment by billziss-gh** [comment]
I cannot test this right now, but try mounting the file system as a drive as well (i.e. add an `X:` at the end of your command line). You can also try the alternative syntax: `--VolumePrefix=\some\test\path`.

Volume prefixes should work under FUSE as they are used widely (e.g. by [SSHFS-Win](https://github.com/winfsp/sshfs-win/blob/master/sshfs-win.c#L205)).

**Comment by pegleGrot** [comment]
Can VolumePrefix be anything or must it be two elements?
I'm mainly thinking `sshfs-win`, and it could be useful if simpler one-part names could be assigned, maybe as an alias.
Not `\\sshfs\user@host`, but just arbitrary custom names like `\\user`.

There is [FSP_MUP_PREFIX_CLASS](https://github.com/winfsp/winfsp/blob/e7c7ccc5c45dc19fbf9f4499ef8104bde89fa488/src/sys/mup.c#L38) to control whether a whole class prefix is registered (`\\class\*`) or not. I'm unsure yet if this would make it easy enough to support single-component UNC paths.
